dash/examples/ex.11.simple-stencil/main.cpp

```diff
@@ -0,0 +1,221 @@
+#include <dash/Init.h>
+#include <dash/Matrix.h>
+#include <dash/Dimensional.h>
+#include <dash/TeamSpec.h>
+#include <dash/algorithm/Fill.h>
+#include <dash/algorithm/Copy.h>
+
+#include <fstream>
+#include <string>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+using element_t = unsigned char;
+using Array_t   = dash::NArray<element_t, 2>;
+using index_t = typename Array_t::index_type;
+
+
+void write_pgm(const std::string & filename, const Array_t & data){
+  if(dash::myid() == 0){
+
+    auto ext_x = data.extent(0);
+    auto ext_y = data.extent(1);
+    std::ofstream file;
+    file.open(filename);
+
+    file << "P2\n" << ext_x << " " << ext_y << std::endl;
+    // Data
+//    std::vector<element_t> buffer(ext_x);
+
+    for(long y=0; y<ext_y; ++y){
+      auto first = data.begin()+ext_x*y; 
+      auto last  = data.begin()+(ext_x*(y+1));
+
+//      BUG!!!!
+//      dash::copy(first, last, buffer.data());
+
+      for(long x=0; x<ext_x; ++x){
+//        file << buffer[x] << " ";
+        file << static_cast<int>(data[x][y]) << " ";
```

Comment:
Yes, of course, but that is extremly slow. I intended to fetch a whole row asynchronously, but there is a bug in dash::copy which asserts that 0 elements will be copied. However I manually checked the GlobIter and the distance is definitly not 0.

Maybe @fuchsto can have a look at. I was not able to reproduce this in a similar unit test.